core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -95,6 +99,12 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
   // The num of current max ExecutorId used to re-register appMaster
   @volatile protected var currentExecutorIdCounter = 0
 
+  // hadoop token manager used by some sub-classes (e.g. Mesos)
+  protected var hadoopDelegationTokenManager: Option[HadoopDelegationTokenManager] = None
+
+  // Hadoop delegation tokens to be sent to the executors.
+  protected var hadoopDelegationCreds: Option[Array[Byte]] = None
```

Comment:
So, another option here is have this be a `val` and have `hadoopDelegationTokenManager` be a `def`. The latter means there's no initialization order issue (so no need for the `lazy val` hack). Since this is really only done once, that should be fine.